job_templates/regenerate_puppet_certificate.erb

```diff
@@ -0,0 +1,46 @@
+<%#
+name: Regenerate puppet certificate
+job_category: Miscellaneous
+snippet: false
+template_inputs:
+- name: Delete certificate on Puppet CA
+  required: true
+  input_type: user
+  description: If yes is selected, Foreman will contact Puppet CA of a host and remove
+    the certificate from there. However this requires safe mode to be disabled.
+  options: "no\r\nyes"
+  advanced: false
+provider_type: SSH
+kind: job_template
+%>
+
+<%#
+name: Regenerate puppet certificate
+job_category: Puppet
+description_format: Regenerate puppet certificate, optionally (with safe mode disabled, it can also remove the certificate from puppet ca)
+snippet: false
+provider_type: SSH
+kind: job_template
+%>
+
+<% # this template requires safe mode to be disabled -%>
+
+<%= render_template('Service Action - SSH Default', :service => 'puppet', :action => 'stop') -%>
+
+<%- if input('Delete certificate on Puppet CA') == 'yes' %>
+# deleting host certificate on puppet ca proxy
+<%-   if preview? -%>
+#   this would delete a certificate on puppet ca, but running in preview mode
+<%-   else -%>
+<%=     @host.puppet_ca_proxy.statuses[:puppetca].destroy(@host.name) && '# deleted cert on proxy' %>
+<%-   end -%>
+<%- end -%>
+
+# delete the old cert and key
+ssl_path=`puppet config print ssldir`
+rm -rf $ssl_path/*
```

Comment:
I think just removing ```$ssl_path``` is good enough since puppet will recreate the directory. This also prevents a ```rm -rf /*``` in case the variable is (accidentally) empty. I'd also quote it just to be sure.